port_ocean/core/utils/utils.py

```diff
@@ -193,24 +193,34 @@ def resolve_entities_diff(
     Returns:
         list[Entity]: Filtered list of source entities, excluding matches found in target
     """
-    target_entities_dict = {}
     source_entities_dict = {}
-    changed_entities = []
 
-    for entity in target_entities:
-        key = (entity.identifier, entity.blueprint)
-        target_entities_dict[key] = entity
+    if (
+        len(
+            [
+                entity
+                for entity in source_entities
+                if entity.is_using_search_identifier or entity.is_using_search_relation
+            ]
+        )
+        > 0
+    ):
+        return source_entities
```

Comment:
```suggestion
if (
            source_entities[0].is_using_search_identifier
            or source_entities[0].is_using_search_relation
        ):
            return source_entities
```
one check is enough, no need full array iteration